### ISSUE TITLE
Problem with Reproducing the Single-Task Results on Cityscapes-Depth

### ISSUE THREAD

**Issue by zhanglijun95** [sunxm2357/AdaShare#9]
Dear Author,

I'm trying to reproduce the single-task results on Cityscapes-Depth, which are presented in Table 2 (Abs.:0.017, Rel.: 0.33, δ < 1:25; 1:252; 1:253: 70.3, 86.3, 93.3). But I cannot get the similar results. Here are some details of my experiments setting:

About model:
1.Use _Deeplab_ResNet_Backbone_ (with parameters BasicBlock, [3, 4, 6, 3] for Deeplab-ResNet34) in your code to produce the feature.
2.Then use the _ASPP architecture_ as head to get the output.

About data:
3. Use the CityScapes dataloader in your code.

About metric:
4. Use the _depth_loss_ and _depth_err_ in your code, and follow the logic in _eval()_ function in _train.py_

About hyperparameters:
5. SGD with lr=0.01, batch size=16, iterations=6,000; then lr=0.0001, iterations=6,000.

Is there anything I miss? Could you please give me any suggestion to get the similar Single-Task results as your experiment? Thank you vey much!

Best Regards,
Lijun

**Comment by J-Wu97** [comment]
> Dear Author,
> 
> I'm trying to reproduce the single-task results on Cityscapes-Depth, which are presented in Table 2 (Abs.:0.017, Rel.: 0.33, δ < 1:25; 1:252; 1:253: 70.3, 86.3, 93.3). But I cannot get the similar results. Here are some details of my experiments setting:
> 
> About model:
> 1.Use _Deeplab_ResNet_Backbone_ (with parameters BasicBlock, [3, 4, 6, 3] for Deeplab-ResNet34) in your code to produce the feature.
> 2.Then use the _ASPP architecture_ as head to get the output.
> 
> About data:
> 3. Use the CityScapes dataloader in your code.
> 
> About metric:
> 4. Use the _depth_loss_ and _depth_err_ in your code, and follow the logic in _eval()_ function in _train.py_
> 
> About hyperparameters:
> 5. SGD with lr=0.01, batch size=16, iterations=6,000; then lr=0.0001, iterations=6,000.
> 
> Is there anything I miss? Could you please give me any suggestion to get the similar Single-Task results as your experiment? Thank you vey much!
> 
> Best Regards,
> Lijun

I meet the same problem when traing single task on the NYUD-V2 dataset.

**Comment by sunxm2357** [comment]
I think you should try Adam instead of SGD. It is pretty useful to get good results on cityscapes
